### PR TITLE
Python: New AI Connector abstract methods

### PR DESCRIPTION
**Pull request by TaoChenOSU**
### Motivation and Context

This PR implements: https://github.com/microsoft/semantic-kernel/blob/main/docs/decisions/0052-python-ai-connector-new-abstract-methods.md (PR: https://github.com/microsoft/semantic-kernel/pull/8430).

### Description

1. Add abstract methods `_inner_get_chat_message_content` and `_inner_get_streaming_chat_message_content` to `ChatCompletionClientBase`.
2. Implement the abstractions in all chat completion connectors.
3. Add abstract methods `_inner_get_text_contents` and `_inner_get_streaming_text_contents` to `TextCompletionClientBase`.
4. Implement the abstractions in all text completion connectors.
5. Remove text completion APIs from `OllamaChatCompletion` (breaking changes).
> No breaking changes on other connectors except Ollama, and public APIs stay the same.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def get_prompt_execution_settings_class(self) -> type["PromptExecutionSettings"]
 
     @override
     @trace_chat_completion(AzureAIInferenceBase.MODEL_PROVIDER_NAME)
-    async def _send_chat_request(
+    async def _inner_get_chat_message_content(
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py:127]: These return lists!
```suggestion
    async def _inner_get_chat_message_contents(
```

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ async def _send_chat_request(
         return [self._create_chat_message_content(response, choice, response_metadata) for choice in response.choices]
 
     @override
-    async def _send_streaming_chat_request(
+    async def _inner_get_streaming_chat_message_content(
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py:147]:
```suggestion
    async def _inner_get_streaming_chat_message_contents(
```

File: python/semantic_kernel/connectors/ai/chat_completion_client_base.py [hunk 3]
```diff
@@ -50,10 +49,9 @@ async def _send_chat_request(
         Returns:
             chat_message_contents (list[ChatMessageContent]): The chat message contents representing the response(s).
         """
-        ...
+        raise NotImplementedError
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/chat_completion_client_base.py:52]: Let's add a message to the error, explaining this change!

**Review by eavanvalkenburg — COMMENTED** [review]
Couple of small notes, but looking great overall, this will make things much easier, thanks!

**Review by moonbox3 — APPROVED** [review]

**Review by alliscode — APPROVED** [review]

**Pull request merged**